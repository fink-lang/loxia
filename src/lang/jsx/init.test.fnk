{fink2js} = import '../../testing/generate.fnk'
{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'


describe 'jsx', fn:
  it 'compiles shorthand', fn:
    expect
      fink2js 'elem = <a/>'
      to_match_snapshot


  it 'compiles with str params', fn:
    expect
      fink2js 'elem = <a foo bar="ni" />'
      to_match_snapshot

    expect
      fink2js "elem = <a foo bar='ni' />"
      to_match_snapshot


  it 'compiles with expr params', fn:
    expect
      fink2js 'elem = <a foo bar={1234} />'
      to_match_snapshot

    expect
      fink2js "elem = <a foo bar='ni' />"
      to_match_snapshot


  it 'compiles empty elem', fn:
    expect
      fink2js 'elem = <b></b>'
      to_match_snapshot


  it 'compiles with children and expr', fn:
    expect
      fink2js 'elem =
        <a>
          foo {ni}
          <b /> ham
          spam
          <c />
          ni
        </a>'
      to_match_snapshot


  it 'compiles fragment', fn:
    expect
      fink2js 'elem = <></>'
      to_match_snapshot


  it 'compiles fragment with children', fn:
    expect
      fink2js 'elem =
        <>
          foo
          <p>bar</p>
        </>'
      to_match_snapshot



describe 'JSX extensions',  fn:
  it 'compiles spread', fn:
    expect
      fink2js '<Foobar spam ...ham shrub />'
      to_match_snapshot


  it "compiles template str attr", fn:
    expect
      fink2js "<Foobar spam='ni: \${1 + 2}' />"
      to_match_snapshot


  it "compiles fink expr as attr values", fn:
    expect
      fink2js "<Foobar spam=ham shrub=1234 ni=-123> foo </Foobar>"
      to_match_snapshot


  it "compiles fink expr with gt comparison", fn:
    expect
      fink2js "<Foobar spam={foo > 123} shrub=1234> ni </Foobar>"
      to_match_snapshot


  it "compiles expr group with call for attr value", fn:
    expect
      fink2js "<Foobar spam=(ham ni) shrub=1234 />"
      to_match_snapshot

