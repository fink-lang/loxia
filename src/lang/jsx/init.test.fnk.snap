// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX extensions compiles class prop as className 1`] = `"<Foobar className=\\"shrub\\" />;"`;

exports[`JSX extensions compiles expr group with call for attr value 1`] = `"<Foobar spam={ham(ni)} shrub={1234} />;"`;

exports[`JSX extensions compiles fink expr as attr values 1`] = `"<Foobar spam={ham} shrub={1234} ni={-123}> foo </Foobar>;"`;

exports[`JSX extensions compiles fink expr with gt comparison 1`] = `"<Foobar spam={foo > 123} shrub={1234}> ni </Foobar>;"`;

exports[`JSX extensions compiles shothand props 1`] = `"<Foobar spam={spam} ham-ni={hamáœ­ni} />;"`;

exports[`JSX extensions compiles spread 1`] = `"<Foobar spam={spam} {...ham} shrub={shrub} />;"`;

exports[`JSX extensions compiles template str attr 1`] = `"<Foobar spam={\`ni: \${1 + 2}\`} />;"`;

exports[`jsx compiles empty elem 1`] = `"export const elem = <b></b>;"`;

exports[`jsx compiles fragment 1`] = `"export const elem = <></>;"`;

exports[`jsx compiles fragment with children 1`] = `
"export const elem = <>
  foo
  <p>bar</p>
</>;"
`;

exports[`jsx compiles hypenate props 1`] = `"export const elem = <a foo-bar=\\"1234\\" />;"`;

exports[`jsx compiles shorthand 1`] = `"export const elem = <a />;"`;

exports[`jsx compiles with children and expr 1`] = `
"export const elem = <a>
  foo {ni}
  <b /> ham
  spam
  <c />
  ni
</a>;"
`;

exports[`jsx compiles with expr params 1`] = `"export const elem = <a foo={foo} bar={1234} />;"`;

exports[`jsx compiles with expr params 2`] = `"export const elem = <a foo={foo} bar=\\"ni\\" />;"`;

exports[`jsx compiles with str params 1`] = `"export const elem = <a foo={foo} bar=\\"ni\\" />;"`;
